components/net/http_loader.rs

```diff
@@ -584,6 +583,16 @@ pub fn http_fetch(
             .unwrap()
             .set_attribute(ResourceAttribute::RequestStart);
 
+        if request.redirect_mode == RedirectMode::Manual ||
+            request.redirect_mode == RedirectMode::Follow
+        {
+            context
+                .timing
+                .lock()
+                .unwrap()
+                .set_attribute(ResourceAttribute::RedirectStart);
```

Comment:
It's not clear to me that this is the right place for this work. The spec says that step 20 of https://w3c.github.io/resource-timing/#processing-model is when the attribute should be set, and that only happens after a redirect has been encountered when processing the response status code.